Src/PCompiler/CompilerCore/Backend/Symbolic/TransformASTPass.cs

```diff
@@ -551,61 +569,62 @@ static private IPStmt HandleReceives(IPStmt statement, Function function, Machin
-                                 rec.CanReceive = function.CanReceive;
-                                 rec.IsNondeterministic = function.IsNondeterministic;
-                                 // make while loop body
-                                 List<IPStmt> loopBody = new List<IPStmt>();
-                                 IEnumerator<IPStmt> bodyEnumerator = loop.Body.Statements.GetEnumerator();
-                                 while (bodyEnumerator.MoveNext())
-                                 {   
-                                     IPStmt stmt = bodyEnumerator.Current;
-                                     IPStmt replaceBreak = ReplaceBreaks(stmt);
-                                     if (replaceBreak != null) {
-                                         loopBody.Add(ReplaceVars(replaceBreak, newVarMap));
-                                     }
-                                 }
-                                 List<VariableAccessExpr> recArgs = new List<VariableAccessExpr>();
-                                 foreach (var param in rec.Signature.Parameters)
-                                 {
-                                     recArgs.Add(new VariableAccessExpr(rec.SourceLocation, param));
-                                 }
-                                 // call the function
-                                 FunCallStmt recCall = new FunCallStmt(loop.SourceLocation, rec, recArgs);
-                                 loopBody = new List<IPStmt>(((CompoundStmt) HandleReceives(new CompoundStmt(rec.SourceLocation, loopBody), rec, machine)).Statements);
-                                 rec.Body = new CompoundStmt(rec.SourceLocation, loopBody);
-                                 if (machine != null) machine.AddMethod(rec);
-                                 // assign local variables
-                                 foreach (var local in function.LocalVariables)
-                                 {
-                                     result.Add(new AssignStmt(local.SourceLocation, new VariableAccessExpr(local.SourceLocation, newVarMap[local]), new VariableAccessExpr(local.SourceLocation, local)));
-                                 }
-                                 // replace the while statement with a function call
-                                 result.Add(recCall);
-                                 if (after != null)
-                                 {
-                                     foreach (var stmt in after.Statements)
-                                     {
-                                         result.Add(ReplaceVars(stmt, newVarMap));
-                                     }
-                                 }
-                                 whileNumber++;
-                                 break;
+                                 throw new NotImplementedException($"Receive in a while statement is not yet supported, found in {machine.Name}");
```

Comment:
Can I request you to create a issue on github that adds a todo that there is a potential bug in the receive in while logic alone with a small repro for the bug that @lmpick can look at. Thanks!

Reply:
Added: Check issue #501